pandas/core/generic.py

```diff
@@ -7806,14 +7791,9 @@ def interpolate(
         fillna_methods = ["ffill", "bfill", "pad", "backfill"]
         if method.lower() in fillna_methods:
```

Comment:
Could you just remove this entire `if` branch? A bad `method` should raise later on

Reply:
Thank you for reviewing this PR. I removed this `if` branch and changed the error message in the `if` branch below. Just one question: we didn't have a test for the previous `ValueError` message:
```
"'fill_value' is not a valid keyword for "
f"{type(self).__name__}.interpolate with method from "
f"{fillna_methods}"
```
Should I add the test?